src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,133 @@
+     * Command parsed by parse() method will be used to identify the query type.
+     * "bye" command will store taskList as a duke.txt file, then return null
+     * which is picked up by the Ui class to shut down the window.
+     *
+     * @param command used to classify the query type from the user.
+     * @param input contains the overall user input.
+     */
+    private String handleQuery(String command, String input) {
+        switch (command) {
+            case "bye":
+                try {
+                    storage.store(tasks);
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+                return null;
+            case "todo":
+                try {
+                    Task t = new Todo(input);
+                    tasks.add(t);
+                    return add(t, tasks.size());
+                } catch (IndexOutOfBoundsException e) {
+                    return new EmptyDescriptionError("todo").getMessage();
+                }
+            case "event":
+                try {
+                    Task t = new Event(input);
+                    tasks.add(t);
+                    return add(t, tasks.size());
+                } catch (IndexOutOfBoundsException e) {
+                    return new EmptyDescriptionError("event").getMessage();
+                }
+            case "deadline":
+                try {
+                    Task t = new Deadline(input);
+                    tasks.add(t);
+                    return add(t, tasks.size());
+                } catch (IndexOutOfBoundsException e) {
+                    return new EmptyDescriptionError("event").getMessage();
+                }
+            case "list":
+                return tasks.list();
+            case "done":
+                Task t = tasks.done(Integer.parseInt(input.substring(5)) - 1);
+                return done(t);
+            case "delete":
+                t = tasks.delete(Integer.parseInt(input.substring(7)) - 1);
+                return delete(t, tasks.size());
+            case "find":
+                String keyword = input.substring(5);
+                return tasks.find(keyword);
+            default:
+                try {
+                    throw new UnknownCommandError("");
+                } catch (UnknownCommandError e) {
+                    return e.getMessage();
+                }
+        }
+    }
+
+    /**
+     * This method will show users that the task has been added successfully to taskList.
+     *
+     * @param t The task being added.
+     * @param size The current size of taskList.
+     */
+    public String add(Task t, int size) {
+        return "     Got it. I've added this task: \n"
+                + String.format("       %s\n", t)
+                + String.format("     Now you have %d tasks in the list.\n", size);
+    }
+
+    /**
+     * This method will show users that the task has been successfully marked as completed.
```

Comment:
Perhaps you could consider beginning the statement with a word like 'returns', 'adds', etc. So for this instance, maybe you could say 'shows users that ...'. Other than that, this sentence makes sense perfectly! I like the crisp nature of your descriptions :)